src/server/game/Handlers/PetHandler.cpp

```diff
@@ -904,5 +904,22 @@ void WorldSession::HandleLearnPreviewTalentsPet(WorldPacket& recvData)
 
 void WorldSession::HandleRequestPetInfoOpcode(WorldPacket& /*recvPacket*/)
 {
-    GetPlayer()->PetSpellInitialize();
+    // Handle the packet CMSG_REQUEST_PET_INFO - sent when player does ingame /reload command
+
+    // Packet sent when player has a pet
+    if (_player->GetPet())
+        _player->PetSpellInitialize();
+
+    if (Unit* charm = _player->GetCharmed())
```

Comment:
is this supposed to be a "else if" or can it be called even if the if "_player->GetPet()" is true ?

Reply:
now looks better :)